app/src/main/java/com/odysee/app/ui/channel/ChannelCommentsFragment.java

```diff
@@ -175,21 +184,40 @@ private void checkAndLoadComments() {
             View commentForm = root.findViewById(R.id.container_comment_form);
             RecyclerView commentsList = root.findViewById(R.id.channel_comments_list);
 
-            if (claim.getTags().contains("disable-comments")) {
-                Helper.setViewVisibility(commentsDisabledText, View.VISIBLE);
-                Helper.setViewVisibility(commentForm, View.GONE);
-                Helper.setViewVisibility(commentsList, View.GONE);
-            } else {
-                Helper.setViewVisibility(commentsDisabledText, View.GONE);
-                Helper.setViewVisibility(commentForm, View.VISIBLE);
-                Helper.setViewVisibility(commentsList, View.VISIBLE);
-                if (commentsList == null || commentsList.getAdapter() == null || commentsList.getAdapter().getItemCount() == 0) {
-                    loadComments();
+            showComments(commentsDisabledText, commentForm, commentsList);
```

Comment:
When comments are disabled, the comment area flickers briefly before it's hidden in favour of the "comments diabled" message. I recommend hiding the comments area before doing the enabled check in order to prevent the flicker.